inc/Subscriptions/BulkSubscriptionSender.php

```diff
@@ -231,9 +231,24 @@ protected function sendList($subscriber_mail, $ids, $ns_id)
         $tlist = '';
         $hlist = '<ul>';
         foreach ($ids as $id) {
+            $pagelog = new PageChangeLog($id);
+            $n = 0;
+            $last = null;
+            do {
+                $rev = $pagelog->getRevisions($n++, 1);
+                $rev = ($rev !== []) ? $rev[0] : null;
+                if ($rev !== null) {
+                    $last = $rev;
+                }
+            } while (!is_null($rev) && $rev > $lastupdate);
+
             $link = wl($id, [], true);
+            $difflink = wl($id, ['do' => 'diff', 'rev' => $last], true);
             $tlist .= '* ' . $link . NL;
-            $hlist .= '<li><a href="' . $link . '">' . hsc($id) . '</a></li>' . NL;
+            $hlist .= '<li>';
+            $hlist .= '<a href="' . $link . '">' . hsc($id) . '</a>';
+            $hlist .= ' (<a href="' . $difflink . '">diff</a>)';
```

Comment:
This should probably use a localized string. We probably already have something suitable in the global $lang array.

I also wonder if the code would be a bit cleaner if the whole inner part of the foreach would be moved into its own method.

Reply:
Done.